app/pages/index.vue

```diff
@@ -84,12 +322,13 @@ function onNextPuzzle() {
           <span class="text-[#60314f]">{{ $t('puzzle.rules.obstacles') }}</span>
         </template>
       </i18n-t>
-      <div class="my-2 flex justify-center" :class="{ invisible: playing }">
+      <div class="my-2 flex justify-center" :class="{ invisible: playing || showScoreSummary }">
         <UButton trailing-icon="i-lucide-arrow-right" size="md" :color="puzzleResult?.success ? 'success' : 'error'" @click="onNextPuzzle">{{ $t('puzzle.next') }}</UButton>
       </div>
     </UContainer>
-    <PuzzleMap v-bind="mapProps" class="mx-auto my-4" @puzzle-completed="onPuzzleCompleted" />
-    <div class="my-2 flex gap-2 justify-center">
+    <PuzzleMap v-if="!showScoreSummary && !showModeSelection" v-bind="mapProps" class="mx-auto my-4" @puzzle-completed="onPuzzleCompleted" />
+      <div class="my-2 flex gap-2 justify-center">
+        <UButton v-if="!isRushMode" icon="i-lucide-zap" size="md" color="warning" @click="toggleRushMode">{{ $t('rushMode.startButton') }}</UButton>
```

Comment:
All buttons in the PR seem to be missing some attribute because they don't enable the hover mouse icon